src/System.Linq.Queryable/tests/QueryFromExpressionTests.cs

```diff
@@ -0,0 +1,508 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Reflection;
+using Xunit;
+
+namespace System.Linq.Tests
+{
+    public class QueryFromExpressionTests
+    {
+        private class SimplePair : IEnumerable<int>
+        {
+            public int First { get; set; }
+
+            public int Second { get; set; }
+
+            public IEnumerator<int> GetEnumerator()
+            {
+                yield return First;
+                yield return Second;
+            }
+
+            IEnumerator IEnumerable.GetEnumerator()
+            {
+                return GetEnumerator();
+            }
+        }
+
+        private static IQueryProvider _prov = Enumerable.Empty<int>().AsQueryable().Provider;
+
+        [Fact]
+        public void ConditionalEqualModuloConstantConstant()
+        {
+            Expression cond = Expression.Condition(
+                Expression.Equal(
+                    Expression.Modulo(Expression.Constant(1), Expression.Constant(2)),
+                    Expression.Constant(0)
+                   ),
+                Expression.Constant(Enumerable.Range(0, 2).AsQueryable()),
+                Expression.Constant(Enumerable.Range(3, 2).AsQueryable())
+            );
+            IQueryable<int> q = _prov.CreateQuery<int>(cond);
+            Assert.Equal(Enumerable.Range(3, 2), q);
+        }
+
+        [Fact]
+        public void ConditionalEqualModuloConstantConstantExplicitConstruction()
+        {
+            Expression cond = Expression.Condition(
+                Expression.Equal(
+                    Expression.Modulo(Expression.Constant(1), Expression.Constant(2)),
+                    Expression.Constant(0)
+                   ),
+                Expression.Constant(Enumerable.Range(0, 2).AsQueryable()),
+                Expression.Constant(Enumerable.Range(3, 2).AsQueryable())
+            );
+            IQueryable<int> q = new EnumerableQuery<int>(cond);
```

Comment:
I think instead I'll just use a simpler expression here. What I intended to test was the expression working when passed to a constructor, whereas of the two ways to turn an expression into an `EnumerableQuery` I use the other (more provider-agnostic) approach in the rest of the tests. I copy-pasted the expression from the previous test, but since that's not what is being tested, it should be simpler.